File: 7.31/Week 4/MVPDagger/app/src/main/java/com/example/jasongomez/mvp_dagger/view/mainactivity/MainActivity.java
package com.example.jasongomez.mvp_dagger.view.mainactivity;

import android.content.Context;
import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;
import android.util.Log;
import android.view.View;

import com.example.jasongomez.mvp_dagger.MyApp;
import com.example.jasongomez.mvp_dagger.R;
import com.example.jasongomez.mvp_dagger.di.MainActivityModule;

import javax.inject.Inject;

public class MainActivity extends AppCompatActivity implements MainActivityContract.View {

    private static final String TAG = "MainActivityTag";

    @Inject
    MainActivityPresenter presenter;

    @Inject
    Context context;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        ((MyApp) getApplication()).getAppComponent().newMainActivityComponent(new MainActivityModule()).inject(this);
        presenter.attachView(this);
        presenter.setContext(context);
    }

    @Override
    public void isPersonSaved(boolean isSaved) {
        Log.d(TAG, "isPersonSaved: " + isSaved);
    }

    @Override
    public void sendPerson(String person) {
        Log.d(TAG, "sendPerson: " + person);
    }

    @Override
    public void showerror(String s) {
        Log.d(TAG, "showerror: ");
    }

    public void doMagic(View view) {
        presenter.savePerson("John Doe");
        presenter.getPerson();
    }
}

File: 7.31/Week 2/kotlin/FragmentCommunicationKotlin/app/src/main/java/com/example/jasongomez/fragmentcommunicationkotlin/MainActivity.kt
package com.example.jasongomez.fragmentcommunicationkotlin

import android.support.v7.app.AppCompatActivity
import android.os.Bundle

class MainActivity : AppCompatActivity(), RecyclerViewFragment.OnPersonSelectedListener {

    private lateinit var recyclerViewFragment: RecyclerViewFragment
    private lateinit var itemFragment: ItemFragment

    companion object {

        private const val RECYCLER_VIEW_FRAGMENT_TAG = "rvFragment"
        private const val ITEM_FRAGMENT_TAG = "itemFragment"
    }

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        recyclerViewFragment = RecyclerViewFragment()
        itemFragment = ItemFragment()

        supportFragmentManager
                .beginTransaction()
                .add(R.id.rvFragContainer, recyclerViewFragment, RECYCLER_VIEW_FRAGMENT_TAG)
                .add(R.id.itemFragContainer, itemFragment, ITEM_FRAGMENT_TAG)
                .commit()
    }


    override fun onPersonSelected(person: Person) {
        val fragment = supportFragmentManager.findFragmentByTag(ITEM_FRAGMENT_TAG) as ItemFragment
        fragment.personSelected(person)
    }
}
